dev/com.ibm.ws.session/src/com/ibm/ws/session/AbstractHttpSessionFacade.java

```diff
@@ -0,0 +1,201 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.session;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Enumeration;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpSession;
+
+import com.ibm.websphere.servlet.session.IBMApplicationSession;
+import com.ibm.websphere.servlet.session.IBMSession;
+import com.ibm.wsspi.servlet.session.IBMSessionExt;
+
+/*
+ * Since Servlet 6.0 - Common APIs 
+ */
+
+public abstract class AbstractHttpSessionFacade implements IBMSession, IBMSessionExt { 
+
+    protected transient AbstractSessionData _session = null;
```

Comment:
@jhanders34  What's liberty's support position here? Should any server level be compatible with another regarding session serialization? No breakage whatsoever? 
- For instance, 17.0.0.4 should still be compatible with 22.0.0.4. 

 Or is it more-so with features and not with server levels? 
 - For instance, all servlet-3.1, or servlet-4.0 should be compatible with each other (regardless of server level). 
 
 @pmd1nh @pnicolucci  Feel free to add any comments in case I'm missing something else.

Reply:
If things are stored in a JCache cache or DB and then you upgrade and a session is refreshed from that DB, we should not break serialization compatibility.  It should be an easy test to write to make sure that we haven't broken things.  When things failover to an updated server when doing a rolling upgrade is the main scenario I am concerned with.

Reply:
Ok I tried serializing SessionData and its subclasses and HttpSessionFacade and its subclasses.  They cannot be serialized today and http session replication does things differently so this is a non issue.  The reason that it doesn't work today is because there is not no argument constructor.  So it can't be deserialized.